src/Init/Data/Cast.lean

```diff
@@ -0,0 +1,101 @@
+/-
+Copyright (c) 2014 Mario Carneiro. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mario Carneiro, Gabriel Ebner
+-/
+prelude
+import Init.Data.Nat.Basic
+import Init.Data.Int.Basic
+
+/-!
+# `NatCast` and `IntCast`
+
+We introduce the typeclass `NatCast R` for a type `R` with a "canonical homomorphism" `Nat → R`.
+The typeclass carries the data of the function, but no required axioms.
+
+This typeclass was introduced to a uniform simp normal form for such morphisms.
```

Comment:
```suggestion
This typeclass provides a uniform `simp` normal form for such morphisms.
```
The existing text wasn't grammatical as far as I could see - but I think it's also better to talk about what it does, rather than why it came to be in the past, in the primary position in reference docs.

Reply:
Or is this the ported library note? In that case, I think it's better to leave the text mostly as is, but indicate this fact higher up. Then I'd say:
```suggestion
This typeclass was introduced to support a uniform `simp` normal form for such morphisms.
```